src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2590,16 +2629,93 @@ public override object PayloadValue(int index)
     {
-        public int Count { get { return GetInt32At(0); } }
-        public int ClrInstanceID { get { return GetInt16At(4); } }
+        /// <summary>
+        /// The time spent on planing the LOH for compaction
+        /// </summary>
+        public int TimePlan { get { return m_timePlan; } }
+
+        /// <summary>
+        /// The time spent on copying objects during LOH compaction
+        /// </summary>
+        public int TimeCompact { get { return m_timeCompact; } }
+
+        /// <summary>
+        /// The time spent on relocating pointers during LOH compaction
+        /// </summary>
+        public int TimeRelocate { get { return m_timeRelocate; } }
+
+        /// <summary>
+        /// The total number of pointers found in the LOH during LOH compaction
+        /// </summary>
+        public long TotalRefs { get { return m_totalRefs; } }
+
+        /// <summary>
+        /// The total number of pointers found pointing to null in the LOH during LOH compaction
+        /// </summary>
+        public long ZeroRefs { get { return m_zeroRefs; } }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            return ToXml(sb).ToString();
+        }
+
+        public StringBuilder ToXml(StringBuilder sb)
+        {
+            sb.Append(" <GCLOHCompactInfo ");
+            TraceEvent.XmlAttrib(sb, "TimePlan", TimePlan);
+            TraceEvent.XmlAttrib(sb, "TimeCompact", TimeCompact);
+            TraceEvent.XmlAttrib(sb, "TimeRelocate", TimeRelocate);
+            TraceEvent.XmlAttrib(sb, "TotalRefs", TotalRefs);
+            TraceEvent.XmlAttrib(sb, "ZeroRefs", ZeroRefs);
+            sb.Append("/>");
+            return sb;
+        }
+
+        #region private
+        internal GCLOHCompactInfo(int timePlan, int timeCompact, int timeRelocate, long totalRefs, long zeroRefs)
+        {
+            m_timePlan = timePlan;
+            m_timeCompact = timeCompact;
+            m_timeRelocate = timeRelocate;
+            m_totalRefs = totalRefs;
+            m_zeroRefs = zeroRefs;
+        }
+
+        private int m_timePlan;
+        private int m_timeCompact;
+        private int m_timeRelocate;
+        private long m_totalRefs;
+        private long m_zeroRefs;
+        #endregion
+    }
+
+    public sealed class GCLOHCompactTraceData : TraceEvent
+    {
+        public int ClrInstanceID { get { return GetInt16At(0); } }
+
+        public int Count { get { return GetInt16At(2); } }
+
+        public GCLOHCompactInfo Info(int heapIndex)
+        {
+            Debug.Assert(0 <= heapIndex && heapIndex < Count);
+            int baseAddress = 4 + SizeOfGCLOHCompactInfo * heapIndex;
+            int timePlan = GetInt32At(baseAddress);
+            int timeCompact = GetInt32At(baseAddress + 4);
+            int timeRelocate = GetInt32At(baseAddress + 8);
+            long totalRefs = (PointerSize == 8) ? GetInt64At(baseAddress + 16) : (long)GetInt32At(12);
```

Comment:
Something doesn't seem right to me here.  Does pointer size affect the size of totalRefs and zeroRefs?  If so, I would expect totalRefs to be at `baseAddress+12` and then zeroRefs to be at `baseAddress+16` or `baseAddress+20`.